### ISSUE TITLE
Applying multiple variants to a single box

### ISSUE THREAD

**Issue by TerrellLinnell** [styled-system/styled-system#1146]
I thought I saw somewhere that I should be able to apply multiple variants to a single box by listing them like so? Currently I only get the rounded border and if I flip them I only get the cursor change.
```
export const Box = styled.div(
	background,
	space,
	color,
	layout,
	flexbox,
	border,
	fontSize,
	fontWeight,
	lineHeight,
	shadow,
	variant({
		variants: {
			rounded: {
				borderRadius: '10px',
			},
			clickable: {
				cursor: 'pointer',
			},
		},
	}),
);

<Box variants={['rounded', 'clickable']}>
```

**Comment by TerrellLinnell** [comment]
I'm currently working around this by using custom prop variants but that will get lengthy and doing it this way seems like it would be useful and succinct.

**Comment by The-Code-Monkey** [comment]
@TerrellLinnell what would be your proposed api for something like this as currently using an array for a prop is a media query. 

I feel like `"rounded, clickable"` would work better?

**Comment by jxnblk** [comment]
With the code above, you can render either `<Box variant='rounded' />` or `<Box variant='clickable' />` -- hope that makes sense

**Comment by TerrellLinnell** [comment]
@The-Code-Monkey Oh yeah I forgot that the array pattern is for media queries, that makes sense. I hadn't thought of an api for it but the way you show would work just fine and would also be useable with the current media query api.

**Comment by The-Code-Monkey** [comment]
@TerrellLinnell I was thinking about this and I think rather than delimit by `,` but use `&&` instead so it follows js conventions a bit more.

so it would be `"rounded&&clickable"`

**Comment by TerrellLinnell** [comment]
@The-Code-Monkey as long as we are able to have spaces this makes sense, I was worried about readability if someone has 6 variants to apply (likely a very small use case) but if the spaces get stripped out along with the && when it gets read by styled system I dont see an issue with code readability.
"rounded && clickable && borderLarge && lorem && ipsum" looks fine to me and does follow js better so it'll be an easier syntax to remember, good idea!

**Comment by The-Code-Monkey** [comment]
Yeah that makes sense as we can just regex for && with or without spaces.

**Comment by TerrellLinnell** [comment]
Alright, I suppose someone needs to change the label to a feature request because I cant edit it or do I need to open a new issue?

**Comment by The-Code-Monkey** [comment]
@TerrellLinnell one question how would it handle multiple variants would the second variant overwrite css keys for the first one if it was `this && that`

**Comment by TerrellLinnell** [comment]
Ideally you wouldn't do that, but realistically someone will, so yes. Also possibly throw a warning about overwriting css keys? I work with @medelman17 who also had some ideas on this and shared this with me
https://github.com/system-ui/theme-ui/pull/586#issuecomment-591206221

I'm thinking this may be a  better solution and where this may actually end up.
```
    <Text variant="h1" modifiers={['bigger', 'bold', 'orange']}>
        Hello!
      </Text>
```

where the variant can be used for the brunt of the styling but then the modifiers (similar to styled-component-modifiers https://github.com/Decisiv/styled-components-modifiers) can be used to fine tune it.

**Comment by storypixel** [comment]
`<Text variant="bigger orange h1">This would be awesome</Text>`

**Comment by switz** [comment]
Any work on this? Multiple variants seems like a really straightforward UX use-case for a library like this.

**Comment by The-Code-Monkey** [comment]
I can take a look into this today see if I can come up with a
valid solution that will work well.

**Comment by RyanNtrce** [comment]
Hi, is there an update on this?
I'm trying to apply multiple variants, but I'm having a hard time figuring out how.

**Comment by mariuslundgard** [comment]
My use case for multiple variants is:
```
<Button color="primary" mode="solid" />
<Button color="primary" mode="ghost" />
```

The first would have a solid blue background, with white text.
The second would have a transparent background, a light blue border, and solid blue text.

This is not possible in `styled-system` afaict?

**Comment by The-Code-Monkey** [comment]
Check out this on my component. https://github.com/The-Code-Monkey/TechStack/blob/dev/packages/components/src/atoms/button/styled.ts you cannot combine them within compose is the only thing I have found

**Comment by The-Code-Monkey** [comment]
This is using my version of styled system which has a few changes. Dunno if it will work in normal styled system

**Comment by seyofori** [comment]
this is probably very messy but.. here's what I came up with to enable multiple variants.. in this case the variant props which the component takes are `variant` and `centered`

```typescript

import {variant as _variant} from 'styled-system'

const APressable = styled(Pressable)<{
  variant: 'primary' | 'secondary';
  centered: boolean;
}>`
  color: white;
  padding: 10px 20px;
  min-height: 50px;

  ${({theme}: {theme: ITheme}) =>
    _variant({
      variants: {
        primary: {
          color: theme.primary,
          backgroundColor: theme.secondary,
        },
        secondary: {
          color: theme.secondary,
          backgroundColor: theme.primary,
        },
      },
    })};

  ${() =>
    _variant({
      prop: 'centered',
      variants: {
        true: {
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
        },
      },
    })}
`;
```
